custom_components/petlibro/devices/device.py

```diff
@@ -18,8 +18,10 @@ def __init__(self, data: dict, api: PetLibroAPI):
 
     def update_data(self, data: dict) -> None:
         """Save the device info from a data dictionary."""
+        _LOGGER.error(f"Updating data with: {data}")
         self._data.update(data)
         self.emit(EVENT_UPDATE)
+        _LOGGER.error(f"Data after update: {self._data}")
```

Comment:
Logger should use `%` and not fstring ([doc](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html)).

```suggestion
        _LOGGER.error("Updating data with: %s", data)
        self._data.update(data)
        self.emit(EVENT_UPDATE)
        _LOGGER.error("Data after update: %s", self._data)
```